bcc/table.go

```diff
@@ -164,6 +164,24 @@ func (table *Table) Delete(keyStr string) error {
 	return nil
 }
 
+// Delete all keys from the table
+func (table *Table) DeleteAll() error {
+	mod := table.module.p
+	fd := C.bpf_table_fd_id(table.module.p, table.id)
+
+	key_size := C.bpf_table_key_size_id(mod, table.id)
+	key := make([]byte, key_size)
+	keyP := unsafe.Pointer(&key[0])
+	res := C.bpf_lookup_elem(fd, keyP, nil)
+	for res = C.bpf_get_next_key(fd, keyP, keyP); res == 0; res = C.bpf_get_next_key(fd, keyP, keyP) {
```

Comment:
bcc now has a function [`bpf_get_first_key`](https://github.com/iovisor/bcc/blob/fdf9b08cee7844d9ad86cbb1ab6b5adeff2b0520/src/cc/libbpf.c#L141) which takes care of handling kernel specific peculiarities. I would prefer we wait for the [update to bcc v0.4.0](https://github.com/iovisor/gobpf/pull/73). After that, you could do the `for` as follows:

```
for res = C.bpf_get_first_key(fd, keyP, keySize); res == 0; res = C.bpf_get_next_key(fd, keyP, keyP) {
```